Core/TypeParser.h

```diff
@@ -20,7 +35,7 @@ class TypeParser {
     /**
      * Parse an encoded type string.
      */
-    static std::vector<std::string> parseEncodedType(const std::string&);
+    static std::vector<ParsedType> parseEncodedType(BinaryNinja::Ref<BinaryNinja::Architecture> arch, const std::string&);
```

Comment:
Architecture objects really don't need to be reference counted as they're singleton objects.